dash.js

```diff
@@ -699,7 +699,20 @@ const MyDash = new Lang.Class({
     _redisplay: function() {
         let favorites = AppFavorites.getAppFavorites().getFavoriteMap();
 
-        let running = this._appSystem.get_running();
+        let running;
+        if (this._dtdSettings.get_boolean('isolate-workspaces')) {
+            running = this._appSystem.get_running().filter(function(_app) {
```

Comment:
The filtering for the running Apps looks a bit messy.
The reason is that we first check if nautilus is being used for Desktop icons. If we don't do this, when a Nautilus window is open on a given workspace (and it's a favorite), it will be shown as active everywhere.